src/util/codec/mysql/json/json_type.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::Json;
+
+const JSON_TYPE_BOOLEAN: &[u8] = b"BOOLEAN";
+const JSON_TYPE_NONE: &[u8] = b"NULL";
+const JSON_TYPE_INTEGER: &[u8] = b"INTEGER";
+const JSON_TYPE_DOUBLE: &[u8] = b"DOUBLE";
+const JSON_TYPE_STRING: &[u8] = b"STRING";
+const JSON_TYPE_OBJECT: &[u8] = b"OBJECT";
+const JSON_TYPE_ARRAY: &[u8] = b"ARRAY";
+
+impl Json {
+    pub fn json_type(&self) -> &[u8] {
```

Comment:
no, it's an attribute function for `json` in sql:  https://dev.mysql.com/doc/refman/5.7/en/json-attribute-functions.html#function_json-type